src/style.js

```diff
@@ -0,0 +1,47 @@
+import {Fill, Stroke, Text, Style as OpenLayersStyle, Circle as CircleStyle} from 'ol/style.js';
+
+const _style = Symbol('style')
+
+class ToolStyle {  
+  
+    constructor(styleOptions) {
+
+      const style = new OpenLayersStyle({
```

Comment:
This class abstracts OpenLayers `Style`, but it still exposes instances of other Openlayers classes (`Stroke`, `Fill`, etc.) via the `getStyle()` method.

How about moving all OpenLayers code into the viewer class to encapsulate the dependency there and using this class only for providing styling attributes (`stroke`, `fill`, `text`, etc.) and setting default values in case they are not provided?

```js

_fill = Symbol('fill')

class ToolStyle {
  constructor(options) {
    this[_fill] = styleOptions.fill !== undefined ? {
      color: 'rgba(0, 255, 0, 0.1)',
      ...
    }
    ...
  }

  get fill() {
    return this[_fill]
  }

  ...
}

```

We may even want to have a separate class for each style type, e.g. `FillStyle`.